js/cloze.js

```diff
@@ -210,7 +210,14 @@
      * @returns {string} Trimmed answer
      */
     this.getUserAnswer = function () {
-      return H5P.trim($input.val());
+      const trimmedAnswer = H5P.trim($input.val().replace(/\&nbsp;/g, ' '));
+      // Set trimmed answer
+      $input.val(trimmedAnswer);
```

Comment:
@icc Pinging you as @ravimajithia is not around anymore and you merged the pull request.

I am not aware what JI-3692 is about (I don't have access to that project :cry:), but this implementation causes some irritation among users.

The `getUserAnswer` function is not only called when actually checking the answer, but also when the state is saved. I just had to investigate a case where users wondered why - when more than one word was expected in the field - spaces that they had just typed would suddenly disappear. That happened, because the state was saved. Well, it's quite an edge case, because the save content state timeout was set to 4 seconds only and disappearing spaces were quite frequent, but I wonder if that should be considered a bug and if I should write an issue for that.

It may come unexpected that a function named `getUserAnswer` actually also sets the value inside the text input field. I assume that there was a particular reason to do this for Wiris on [H5P.com](http://h5p.com/) (?), but it's also done on regular text input fields. Potentially, a separate function that trims the value and uses `setUserInput` (or maybe all inside setUserInput) could be used and only called when the answers are checked, but not when saving the state.